Recrutify/Recrutify.DataAccess/Repositories/UserRepository.cs

```diff
@@ -51,18 +51,50 @@ public async Task<IEnumerable<User>> GetByRoles(IEnumerable<Role> roles)
         }
 
         public Task BulkAddProjectRolesAsync(Guid projectId, IDictionary<Guid, IEnumerable<Role>> usersRoles)
+        {
+            var updateModels = GetUpdateModelByBulkAddRoles(projectId, usersRoles);
+            return GetCollection().BulkWriteAsync(updateModels);
+        }
+
+        public Task BulkUpdateProjectRolesAsync(Guid projectId, IDictionary<Guid, IEnumerable<Role>> newUsersRoles, IDictionary<Guid, IEnumerable<Role>> removeUsersRoles, IDictionary<Guid, IEnumerable<Role>> updateUsersRoles)
+        {
+            var binaryProjectId = new BsonBinaryData(projectId, GuidRepresentation.Standard);
+            var arrayFilters = new List<ArrayFilterDefinition>
+            {
+               new BsonDocumentArrayFilterDefinition<ProjectResult>(new BsonDocument("keyValuePair.k", binaryProjectId)),
+            };
+            var updateBuilder = Builders<User>.Update;
+
+            var updateModelsWithNewUsers = GetUpdateModelByBulkAddRoles(projectId, newUsersRoles);
+
+            var updateModelsWithRemoteUsers = removeUsersRoles.Select(ur => new UpdateOneModel<User>(
+                                                    _filterBuilder.Eq(u => u.Id, ur.Key),
+                                                    updateBuilder
+                                                    .Pull(
+                                                       $"{nameof(User.ProjectRoles)}.$[keyValuePair].v",
```

Comment:
What if you try remove not just value (only roles), but roles with project key, to not just leave project if with empty roles. Google some examples